aries/aries_vcx_core/src/wallet/indy/mod.rs

```diff
@@ -133,6 +81,23 @@ pub struct WalletRecord {
     tags: Option<String>,
 }
 
+impl WalletRecord {
```

Comment:
In the codebase, there is

* `Record` struct
* `WalletRecord` trait
* `RecordWallet` trait
* `IndyRecord` struct which is a name for `RecordStruct` in `indy_api_types`

now with all kinds of conversions between them. This is quite confusing if you ask me as a reader. If there is no way to eradicate some of those, we should at least find better names.

Reply:
I could remove `RecordWallet` by squashing it together with `DidWallet` into `BaseWallet` if you prefer. 

`Record` was recently introduced to `aries_vcx_core` as an effort to improve the types, so I think this one will stay. 

As for the others, I do not see them going away as long as we keep the code that is using them. `WalletRecord` is returned as a json string from methods in `api_global`, so that is effectively a part of public API exposed by that crate. `Record` from `indy_api_types` is used in `libvidrtools`, I imagine this one will go away as we continue to refactor.

Reply:
Nah, I would not squash them. Considering alternative names for `RecordWallet`, e.g. `CRUDWallet`, `DataWallet` comes to mind, but none seems preferable.

Reply:
> Record struct
> WalletRecord trait
> RecordWallet trait
> IndyRecord struct which is a name for RecordStruct in indy_api_types

`IndyRecord` is not actually type/struct defined, but only used as alias
```
use indy_api_types::domain::wallet::Record as IndyRecord;
```

Perhaps we wouldn't need to use alias if we have better names for the remaining 3:


> Record struct 
> WalletRecord trait
> RecordWallet trait

- `WalletRecord` is not trait but struct, but  there's actually 2 of them:
  - one defined in `wallet/indy/mod.rs`
  - another one defined in `aries/misc/legacy/libvdrtools/indy-wallet/src/lib.rs`
I would keep the one inside `libvdrtools` as it is, it's local to libvdrtools. But we could rename `wallet/indy/mod.rs`, as that's seems to be something we defined, something that we made indy specific. It's usage seems to be narrowly limited in `libvcx` specifically for method `wallet_get_wallet_record`. If we don't want to remove it altogether in this PR to avoid making unnecessary changes, we could rename this to `IndyWalletRecord`. Ultimately remove it altogether later. 

- `Record` 
there's also 2 of this, one our ours, and one in `libvdrtools` - which is why we then consequently do
```
use indy_api_types::domain::wallet::Record as IndyRecord;
```
We could rename the `Record` inside `libvdrtools` to:
1. avoid this aliasing
2. avoid conflicts and being explicit about the fact it's coupled with indy/vdrtools.

- `RecordWallet` trait can stay IMO as is, it fits nice with the other trait name -`DidWallet`

Reply:
Rename sounds ok to me, I think some of these structs will be dropped in time as we continue to refactor.